v2_papers/2009/kind_histogram.json

```diff
@@ -0,0 +1,179 @@
+    "year": 2009,
+    "organization_publishers": [
+      "ieee"
+    ],
+    "pages_number_of": 12,
+    "bibtex": {
+      "type": "article",
+      "volume": "6",
+      "issue": "2",
+      "pages": "110--121"
+    },
+    "access_open": false,
+    "curated_by": "Meghdouri, f.",
+    "curated_last_revision": "11-04-2018",
+    "curated_revision_number": 2
+  },
+  "data": {
+    "datasets": [
+      {
+        "name": "campus of IBM Research",
+        "types": [],
+        "generation_year": 2004,
+        "details": [
+          "raw"
+        ],
+        "subsets": [],
+        "availability": "private",
+        "format": "packet",
+        "generation": "captured"
+      },
+      {
+        "name": "data center",
+        "types": [],
+        "generation_year": 2007,
+        "details": [
+          "raw"
+        ],
+        "subsets": [],
+        "availability": "private",
+        "format": "packet",
+        "generation": "captured",
+        "covered_period": "months"
+      },
+      {
+        "name": "DARPA IDeval",
+        "availability": "public",
+        "format": "packet",
+        "types": [],
+        "generation": "altered_captured",
+        "generation_year": 1999,
+        "covered_period": "weeks",
+        "details": [
+          "raw"
+        ],
+        "subsets": []
+      }
+    ]
+  },
+  "preprocessing": {
+    "performed_feature_selection": true,
+    "packet_analysis_oriented": false,
+    "flow_analysis_oriented": true,
+    "flow_aggregation_analysis_oriented": false,
+    "tools": [],
+    "normalization_type": "no",
+    "transformations": [
+      "_histograms",
+      "_metric_space",
+      "entropy"
+    ],
+    "final_data_format": "_histograms",
+    "feature_selections": [
+      {
+        "name": "constant-removal",
+        "type": "feature_reduction",
+        "classifier": "none",
+        "role": "main"
+      },
+      {
+        "name": "pca",
+        "type": "feature_reduction",
+        "classifier": "none",
+        "role": "main"
+      }
+    ],
+    "flows": [
+      {
+        "selection": "study_based",
+        "role": "main",
+        "main_goal": "anomaly_detection",
+        "active_timeout": "missing",
+        "idle_timeout": "missing",
+        "features": [
+          "sourceIPv4Address",
+          "destinationIPv4Address",
+          "sourceTransportPort",
+          "destinationTransportPort",
+          "protocolIdentifier",
+          "octetTotalCount",
+          "packetTotalCount",
+          "flowDurationMilliseconds"
+        ],
+        "key_features": []
+      }
+    ]
+  },
+  "analysis_method": {
+    "supervised_learning": true,
+    "unsupervised_learning": true,
+    "semisupervised_learning": false,
+    "anomaly_detection": true,
+    "tools": [],
+    "algorithms": [
+      {
+        "name": "hierarchical cluster analysis",
+        "subname": "hca",
+        "learning": "unsupervised",
+        "role": "intermediate",
```

Comment:
Hierarchical clustering is the ``main`` method in this paper (not ``intermediate``). Actually, even though they mention k-means, I can't find where they use it.

Reply:
That is true, I was confused and I thought they used k-means for clustering and hca for classification.